### PR TITLE
All mobile platforms go down the RunTests.sh path, not just browser.

### LINKED ISSUE #107551: [android][infra] Runtime tests on Android are not getting sent to Helix
Some recent changes likely from this range https://github.com/dotnet/runtime/compare/516e81e8...57a4b043 caused that Android runtime tests are not properly bundled and sent Helix (`SendHelixJob given no WorkItems to send.`)

Affected CI:
- `android-x64 Release AllSubsets_Mono_RuntimeTests minijit`
- `android-x64 Release AllSubsets_Mono_RuntimeTests_Interp monointerpreter`

image

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fixed by doing the same change that restored WASM test runs for Android.

Restores running runtime tests on Android.

Fixes #107551

### CODE REVIEW

File: src/tests/Common/helixpublishwitharcade.proj [hunk 1]
```diff
@@ -336,8 +336,8 @@
       <_MergedWrapperMarker Include="$(TestBinDir)**\*.MergedTestAssembly" Exclude="$(TestBinDir)**\supportFiles\*.MergedTestAssembly" />
 
       <_MergedWrapperMarker Update="@(_MergedWrapperMarker)">
-        <TestExecutionScriptPath Condition="'$(TargetsBrowser)' != 'true'">$([System.IO.Path]::ChangeExtension('%(Identity)', '.$(TestScriptExtension)'))</TestExecutionScriptPath>
-        <TestExecutionScriptPath Condition="'$(TargetsBrowser)' == 'true'">%(RootDir)%(Directory)AppBundle/RunTests.$(TestScriptExtension)</TestExecutionScriptPath>
+        <TestExecutionScriptPath Condition="'$(TargetsMobile)' != 'true'">$([System.IO.Path]::ChangeExtension('%(Identity)', '.$(TestScriptExtension)'))</TestExecutionScriptPath>
+        <TestExecutionScriptPath Condition="'$(TargetsMobile)' == 'true'">%(RootDir)%(Directory)AppBundle/RunTests.$(TestScriptExtension)</TestExecutionScriptPath>
```
Comment (matouskozak) [inline src/tests/Common/helixpublishwitharcade.proj:340]: Is `TargetsBrowser` a subset of `TargetsMobile` in this case?

**Review by matouskozak — COMMENTED** [review]
Thanks a lot for fixing the issue!

File: src/tests/Common/helixpublishwitharcade.proj (hunk 1 above)
Comment (jkoritzinsky) [reply src/tests/Common/helixpublishwitharcade.proj:340]: Yep, we've traditionally considered browser a "mobile" OS. `TargetsMobile` effectively means "can't run as a traditional desktop app".

**Review by matouskozak — APPROVED** [review]
Looking good, let's wait if the android-x64 (emulator) runtime tests pass.

**Comment by jkoritzinsky** [comment]
Looks like the tests are running now but there's some failures in the Interop test suite. I'll investigate that as well before merging as it's the only failure.

**Comment by jkoritzinsky** [comment]
Failures are the usual ADB failures of various flavors. This is ready for merge.

**Pull request merged**